modules/apps/petra/petra-url-pattern-mapper/src/main/java/com/liferay/petra/url/pattern/mapper/URLPatternMapper.java

```diff
@@ -14,11 +14,34 @@
 
 package com.liferay.petra.url.pattern.mapper;
 
+import java.util.HashSet;
+import java.util.Set;
+import java.util.function.Consumer;
+
 /**
  * @author Carlos Sierra Andrés
  */
 public interface URLPatternMapper<T> {
 
+	public void consumeValues(String urlPath, Consumer<T> consumer);
+
 	public T getValue(String urlPath);
 
+	public default Set<T> getValues(String urlPath) {
+		Set<T> values = new HashSet<>(Long.SIZE);
+
+		Consumer<T> consumer = new Consumer<T>() {
+
+			@Override
+			public void accept(T t) {
+				values.add(t);
+			}
+
+		};
+
+		consumeValues(urlPath, consumer);
```

Comment:
@arthurchan35 can't this be a `values::add`?

Reply:
instead of creating a full consumer... just a method reference to `values::add`?